cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java

```diff
@@ -108,12 +110,17 @@ private void updateIndex(String indexName, List<FieldSort> fieldNames) throws Qu
 
         Changes changes;
         long lastSequence = sequenceNumberForIndex(indexName);
-
-        do {
-            changes = database.changes(lastSequence, 10000);
-            updateIndex(indexName, fieldNames, changes, lastSequence);
-            lastSequence = changes.getLastSequence();
-        } while (changes.size() > 0);
+        try {
+            do {
+                changes = database.changes(lastSequence, 10000);
+                updateIndex(indexName, fieldNames, changes, lastSequence);
+                lastSequence = changes.getLastSequence();
+            } while (changes.size() > 0);
+        } catch (DocumentStoreException e) {
+            String message = String.format(Locale.ENGLISH, "Failed to get changes feed from %d", lastSequence);
+            logger.log(Level.SEVERE, message, e);
+            throw new QueryException(e.getCause());
```

Comment:
The query code shouldn't call the public API so there should be no `DocumentStoreException` in the first place. In the absence of an even bigger refactor, this is the best we can do. There's no loss if information from skipping out this exception.